### DISCUSSION TITLE
Pagination testing doesn't seem to work properly.

### DISCUSSION THREAD

**Discussion by bzy107** [filamentphp/filament#9246] [Help] [answered]
I've written a test to ensure that the 2nd page is displayed as follows:

```php
test('can render 2nd page', function () {
    $items = Item::factory()->count(30)->create();

    livewire(ItemResource\Pages\ListItems::class)
        ->set('page', 2)
        ->assertCanSeeTableRecords($items->range(11, 20));
});
```

I want to test that the 2nd page is displayed, and I'm referencing the following documentation, which advises using `set('page', 2)`:
> If your table uses pagination, assertCanSeeTableRecords() will only check for records on the first page. To switch pages, call set('page', 2).
https://filamentphp.com/docs/3.x/tables/testing#render

However, when I inspect the error HTML, it shows "Showing 1 to 10," suggesting that the 2nd page is not being displayed. 
How can I write a test to verify that the 2nd page is displayed?

**Comment by sinnbeck** [comment, marked as answer]
Try `->call('gotoPage', 2, 'page')`

Also be sure that the $items is in the exact same order as expected. If so sort by default by created at the order wont be correct as they all will have the exact same created_at timestamp

**Reply by sinnbeck** [reply]
It seems you can also do
```
->set('paginators.page', 2)

**Reply by bzy107** [reply]
Thank you. It's been resolved.

```php
test('can render 2nd page', function () {
    $items = Item::factory()->count(30)->create();

    livewire(ItemResource\Pages\ListItems::class)
        ->set('paginators.page', 2)
        ->assertCanSeeTableRecords($items->skip(10)->take(10));
});
```
